### PR TITLE
Account Number redesign

### LINKED ISSUE #303: Add visual indicator that user can copy the address by tapping on it in the side menu
Expected Behavior
There is a visual indication that user can copy the account key from the side menu. Should be a copy icon or just say in plain text "Tap to copy"

Current Behavior
Users are not aware that they can copy the account key by tapping over it. As a result, they type it in manually and do mistakes > send the funds to non-existing addresses. More context here: https://celo-org.slack.com/archives/C0134SBQTQR/p1619797847006600

- [ ] Add visual indicator according to the design specs created here: https://app.zenhub.com/workspaces/wallet-backlog-board-60092d006a84e3001d0d5936/issues/celo-org/wallet/302

- [ ] Add analytics event when users tap over account number: `account_number_copied`

### PR DESCRIPTION
**Pull request by isabellewei**
### Description

Update Account Number design, add a visual "Tap to Copy" indicator

### Tested

Screenshot_1622570174
Screenshot_1622570318
Screenshot_1622945840

### Related issues

- Fixes #303

### CODE REVIEW

File: packages/mobile/src/components/AccountNumber.tsx [hunk 1]
```diff
@@ -41,39 +41,42 @@ export default function AccountNumber({ address, touchDisabled, location }: Prop
   // Turns '0xce10ce10ce10ce10ce10ce10ce10ce10ce10ce10'
   // into 'ce10 ce10 ce10 ce10 ce10 ce10 ce10 ce10 ce10 ce10'
   const addressChunks = getAddressChunks(address)
+  const addressString = '0x ' + addressChunks.reduce((x, y) => x + ' ' + y)
```
Comment (gnardini) [inline packages/mobile/src/components/AccountNumber.tsx:44]: This didn't have any tests but it'd probably be good to add some :D Maybe just checking that the address is populated correctly with the right spaces and that on press the `Clipboard.setString` function gets called with the right parameter?

**Review by gnardini — COMMENTED** [review]
Looks great!! 🚀

File: packages/mobile/src/transactions/UserSection.tsx [hunk 2]
```diff
@@ -74,8 +74,9 @@ export default function UserSection({
             </Text>
           )}
           <View style={styles.accountBox}>
-            <Text style={styles.accountLabel}>{t('accountNumberLabel')}</Text>
-            <AccountNumber address={address || ''} location={Screens.TransactionReview} />
+            <Text style={styles.accountLabel}>{t('dappkit:address')}</Text>
```
Comment (jeanregisser) [inline packages/mobile/src/transactions/UserSection.tsx:77]: We should keep `accountNumberLabel` and update the locale instead.

**Review by jeanregisser — CHANGES_REQUESTED** [review]
Nice 👍 

Is the "payment sent" screenshot in the PR outdated? According to the Figma design, the address shouldn't be limited to 5 chunks in the first line.
Screenshot 2021-06-03 at 12 12 34

**Comment by isabellewei** [comment]
@jeanregisser thanks, I've fixed it 🙏

**Review by jeanregisser — APPROVED** [review]
Nice, ship it 👍

**Comment by MuckT** [comment]
@isabellewei looks like these changes impacted the cash in / cash out cryptocurrency exchange account number.

Simulator Screen Shot - iPhone 8 - 2021-06-10 at 12 43 08

Applying the below styles `AccountNumber.tsx` appears to fix the issue on iOS and Android.
```JavaScript
return touchDisabled ? (
    <Text style={styles.text}>{addressString}</Text>
  ) : (
    <TouchableOpacity
      onLongPress={onPressAddress}
      onPress={onPressAddress}
      testID="CopyAddressToClipboard"
      style={{flexShrink:1}}
    >
      <Text style={styles.text}>{addressString}</Text>
    </TouchableOpacity>
  )
```

**Comment by Celoqa** [comment]
Hi @MuckT Again We have cross verified this issue on Latest android build V1.16.1 & Testflight build V1.16.1(67) and updated the status in testqulaity tool.

**Testquality tool link:**

**Android:**

- https://valora.testquality.com/project/12128/plan/20854/run/148956/result/2432161
**iOS:**

- https://valora.testquality.com/project/12128/plan/20854/run/148956/result/2432189

And also we have tested #482 


**Devices:** Vivo v11 pro (10), iPhone 6+(12.4.5)

CC: @isabellewei

**Pull request merged**